graspy/inference/latent_distribution_test.py

```diff
@@ -14,121 +14,158 @@
+
+        if not isinstance(metric, str) and not callable(metric):
+            msg = "Metric must be str or callable, not {}".format(type(metric))
+            raise TypeError(msg)
+        elif metric not in _VALID_METRICS and not callable(metric):
+            msg = "Unknown metric {}. Valid metrics are {}, or a callable".format(
+                metric, _VALID_METRICS
+            )
+            raise ValueError(msg)
+
         if n_components is not None:
             if not isinstance(n_components, int):
-                msg = "n_components must an int, not {}.".format(type(n_components))
+                msg = "n_components must be an int, not {}.".format(type(n_components))
                 raise TypeError(msg)
 
         if not isinstance(n_bootstraps, int):
-            msg = "n_bootstraps must an int, not {}".format(type(n_bootstraps))
+            msg = "n_bootstraps must be an int, not {}".format(type(n_bootstraps))
             raise TypeError(msg)
         elif n_bootstraps < 1:
             msg = "{} is invalid number of bootstraps, must be greater than 1"
             raise ValueError(msg.format(n_bootstraps))
 
-        if bandwidth is not None and not isinstance(bandwidth, float):
-            msg = "bandwidth must an int, not {}".format(type(bandwidth))
+        if not isinstance(workers, int):
+            msg = "workers must be an int, not {}".format(type(workers))
             raise TypeError(msg)
+        elif workers <= 0:
+            msg = "{} is invalid number of workers, must be greater than 0"
+            raise ValueError(msg.format(workers))
 
         super().__init__(embedding="ase", n_components=n_components)
+
+        if callable(metric):
+            metric_func = metric
+        else:
+            if metric == "gaussian":
+                metric_func = _medial_gaussian_kernel
+            else:
+
+                def metric_func(X, Y=None, metric=metric, workers=None):
+                    return pairwise_distances(X, Y, metric=metric)
+
+        self.test = KSample(test, compute_distance=metric_func)
         self.n_bootstraps = n_bootstraps
-        self.bandwidth = bandwidth
-
-    def _gaussian_covariance(self, X, Y):
-        diffs = np.expand_dims(X, 1) - np.expand_dims(Y, 0)
-        if self.bandwidth is None:
-            self.bandwidth = 0.5
-        return np.exp(-0.5 * np.sum(diffs ** 2, axis=2) / self.bandwidth ** 2)
-
-    def _statistic(self, X, Y):
-        N, _ = X.shape
-        M, _ = Y.shape
-        x_stat = np.sum(self._gaussian_covariance(X, X) - np.eye(N)) / (N * (N - 1))
-        y_stat = np.sum(self._gaussian_covariance(Y, Y) - np.eye(M)) / (M * (M - 1))
-        xy_stat = np.sum(self._gaussian_covariance(X, Y)) / (N * M)
-        return x_stat - 2 * xy_stat + y_stat
+        self.workers = workers
 
     def _embed(self, A1, A2):
+        if not is_symmetric(A1) or not is_symmetric(A2):
+            raise NotImplementedError()  # TODO asymmetric case
```

Comment:
reminder: this should specify "Currently, testing is only supported for undirected graphs."

Reply:
fixed.